packages/core/src/Serialization/Common/ExUnits.ts

```diff
@@ -0,0 +1,159 @@
+   *
+   * @returns The ExUnits in CBOR format.
+   */
+  toCbor(): HexBlob {
+    const writer = new CborWriter();
+
+    if (this.#originalBytes) return this.#originalBytes;
+
+    // ex_units = [mem: uint, steps: uint]
+    writer.writeStartArray(EX_UNITS_ARRAY_SIZE);
+    writer.writeInt(this.#mem);
+    writer.writeInt(this.#steps);
+
+    return writer.encodeAsHex();
+  }
+
+  /**
+   * Deserializes the ExUnits from a CBOR byte array.
+   *
+   * @param cbor The CBOR encoded ExUnits object.
+   * @returns The new ExUnits instance.
+   */
+  static fromCbor(cbor: HexBlob): ExUnits {
+    const reader = new CborReader(cbor);
+
+    const length = reader.readStartArray();
+
+    if (length !== EX_UNITS_ARRAY_SIZE)
+      throw new InvalidArgumentError(
+        'cbor',
+        `Expected an array of ${EX_UNITS_ARRAY_SIZE} elements, but got an array of ${length} elements`
+      );
+
+    const mem = reader.readUInt();
+    const steps = reader.readUInt();
+
+    reader.readEndArray();
+
+    const exUnit = new ExUnits(mem, steps);
+    exUnit.#originalBytes = cbor;
+
+    return exUnit;
+  }
+
+  /**
+   * Creates a Core ExUnits object from the current ExUnits object.
+   *
+   * @returns The Core ExUnits object.
+   */
+  toCore(): Cardano.ExUnits {
+    return {
+      memory: Number(this.#mem),
+      steps: Number(this.#steps)
+    };
+  }
+
+  /**
+   * Creates a ExUnits object from the given Core ExUnits object.
+   *
+   * @param exUnits core ExUnits object.
+   */
+  static fromCore(exUnits: Cardano.ExUnits) {
+    return new ExUnits(BigInt(exUnits.memory), BigInt(exUnits.steps));
+  }
+
+  /**
+   * Gets the amount of memory the script execution will consume.
+   *
+   * @returns The amount of memory (in bytes).
+   */
+  mem(): bigint {
+    return this.#mem;
+  }
+
+  /**
+   * Sets the amount of memory the script execution will consume.
+   *
+   * @param mem The amount of memory (in bytes).
+   */
+  setMem(mem: bigint): void {
+    this.#mem = mem;
+    this.#originalBytes = undefined;
+  }
+
+  /**
+   * Gets the number of CPU steps that running ths script will take.
+   *
+   * @returns The number of CPU steps.
+   */
+  steps(): bigint {
+    return this.#steps;
+  }
+
+  /**
+   * Sets the number of CPU steps that running ths script will take.
+   *
+   * @param steps The number of CPU steps.
+   */
+  setSteps(steps: bigint): void {
+    this.#steps = steps;
+    this.#originalBytes = undefined;
+  }
+
+  /**
+   * Adds the memory and cpu steps of the current ExUnits and the given ExUnits and returns the result.
+   *
+   * @param other The other ExUnits to be added with this ExUnits.
+   */
+  add(other: ExUnits): ExUnits {
+    const mem = this.#mem + other.#mem;
+    const steps = this.#steps + other.#steps;
+
+    return new ExUnits(mem, steps);
+  }
+
+  /**
+   * Used to create a dummy ExUnits that takes up the maximum size possible in cbor to provide an upper bound on tx size.
+   */
+  static dummy(): ExUnits {
```

Comment:
Why do we need this?

Reply:
Yeah in that case let's remove it for now, we can re-add it as test util or export from `util-dev` package if/when needed